src/mpi/init/initthread.c

```diff
@@ -248,6 +248,19 @@ static int thread_cs_init(void)
 
     MPID_THREADPRIV_KEY_CREATE;
 
+    {
+        /*
+         * Trick the Intel compiler into treating MPIR_Per_thread as TLS
+         * See https://github.com/pmodels/mpich/issues/3437
```

Comment:
We might as well call it a compiler bug, how about  ` * Hack to a Intel compiler bug on mac OS skipping TLS storage without usage`?

Reply:
I agree. Will explicit mention this is a bug.